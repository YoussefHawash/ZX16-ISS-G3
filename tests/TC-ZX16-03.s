# x1 =10   0000 0000 0000 1010
# x2 = 3   0000 0000 0000 0011 
# x3 = 2   0000 0000 0000 0010
# x4 = -16 1111 1111 1111 0000
Add x1 ,x2  #x1=13    0000000000001101  
SLTU x2 ,x1  #x2 =0000 0000 0000 0001
NOT x2        #x2 = 1111 1111 1111 1100  65532
SLL x2 ,x1    #x2=  1100 0000 0000 0000  49152
LI x3,-16
SRAI x3, 2    #x3 = 0000 0000 0000 0000  0
OR  x1 , x3   #x1=  0000 0000 0000 1101 13
OR x3, x2  	#x3 =1100 0000 0000 1101 49152
XOR x2, x4	#x2 = 0011 1111 1111 1101 16381
MV  x5 ,x1      #x5 = 0000 0000 0000 1101 13
ECALL 0x008
# x0 =0 x1= 13 x2= 16381 x3=49152 x4= -16 x5 =13
ECALL 0x00A
