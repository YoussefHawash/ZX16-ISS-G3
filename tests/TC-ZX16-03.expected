SLTU x2, x1
XORI x2, 0x7F
SLL x2, x1
LI x3, 0x70
SRAI x3, 0x02
OR x1, x3
OR x3, x2
XOR x2, x4
MV x5, x1
ECALL 8    # Registers Dump
ECALL 10    # Program Exit